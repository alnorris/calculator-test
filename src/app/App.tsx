import React from 'react';
import ButtonGrid from '../components/ButtonGrid'
import NumDisplay from '../components/NumDisplay'
import styled from 'styled-components'
import useCalcProcessor from '../modules/useCalcProcessor'

const CalculatorWrapper = styled.div`
  margin: 0 auto;
  display: flex;
  width: 240px;
  flex-direction: column;
`

const App: React.FC = () => {

  const { onKey, numDisplayVal } = useCalcProcessor()

  return (
    <CalculatorWrapper>
      <NumDisplay>{numDisplayVal}</NumDisplay>
      <ButtonGrid onChange={onKey} />
    </CalculatorWrapper>
  )
}

export default App
