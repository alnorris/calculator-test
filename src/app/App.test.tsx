import React from 'react'
import { shallow, mount } from 'enzyme'
import App from './App'

jest.doMock('../modules/useCalcProcessor', () => {
  return () => ({
    onKey: jest.fn(),
    numDisplayVal: 88
  })
})

describe('App', () => {
  beforeEach(() => jest.resetModules())
  it('Render app without crashing', () => {
    shallow(<App />)
  })

  it('Test display initial 0', async () => {

    const wrapper = mount(<App />)
    const x = wrapper.find('numDisplay').text()
    expect(x).toBe("0")
  })

  it('Test display 9', async () => {

    const wrapper = mount(<App />)
    wrapper.find('button[children="9"]').simulate("click")

    const x = wrapper.find('numDisplay').text()
    expect(x).toBe("9")
  })

  it('Test display clear', async () => {

    const wrapper = mount(<App />)
    wrapper.find('button[children="9"]').simulate("click")
    wrapper.find('button[children="C"]').simulate("click")

    const x = wrapper.find('numDisplay').text()
    expect(x).toBe("0")
  })

  it('Test display negative', async () => {

    const wrapper = mount(<App />)
    wrapper.find('button[children="8"]').simulate("click")
    wrapper.find('button[children="-/+"]').simulate("click")

    const x = wrapper.find('numDisplay').text()
    expect(x).toBe("-8")
  })


  it('Test display equals', async () => {

    const wrapper = mount(<App />)
    wrapper.find('button[children="8"]').simulate("click")
    wrapper.find('button[children="+"]').simulate("click")
    wrapper.find('button[children="8"]').simulate("click")
    wrapper.find('button[children="8"]').simulate("click")

    const x = wrapper.find('numDisplay').text()
    expect(x).toBe("88")
  })

})