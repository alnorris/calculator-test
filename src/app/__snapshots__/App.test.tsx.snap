// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Add 2+2 1`] = `
<App>
  <styled.div>
    <div
      className="sc-AxhUy iLDogk"
    >
      <styled.div>
        <div
          className="sc-AxhCb iveHrx"
        >
          0
        </div>
      </styled.div>
      <ButtonGrid
        onChange={[Function]}
      >
        <styled.div
          onClick={[Function]}
        >
          <div
            className="sc-AxiKw jCymji"
            onClick={[Function]}
          >
            <styled.button>
              <button
                className="sc-AxjAm kVIsjG"
              >
                C
              </button>
            </styled.button>
            <styled.button>
              <button
                className="sc-AxjAm kVIsjG"
              >
                -/+
              </button>
            </styled.button>
            <styled.button>
              <button
                className="sc-AxjAm kVIsjG"
              >
                %
              </button>
            </styled.button>
            <styled.button
              isOperator={true}
            >
              <button
                className="sc-AxjAm ighVHx"
              >
                /
              </button>
            </styled.button>
            <styled.button>
              <button
                className="sc-AxjAm kVIsjG"
              >
                9
              </button>
            </styled.button>
            <styled.button>
              <button
                className="sc-AxjAm kVIsjG"
              >
                8
              </button>
            </styled.button>
            <styled.button>
              <button
                className="sc-AxjAm kVIsjG"
              >
                7
              </button>
            </styled.button>
            <styled.button
              isOperator={true}
            >
              <button
                className="sc-AxjAm ighVHx"
              >
                x
              </button>
            </styled.button>
            <styled.button>
              <button
                className="sc-AxjAm kVIsjG"
              >
                6
              </button>
            </styled.button>
            <styled.button>
              <button
                className="sc-AxjAm kVIsjG"
              >
                5
              </button>
            </styled.button>
            <styled.button>
              <button
                className="sc-AxjAm kVIsjG"
              >
                4
              </button>
            </styled.button>
            <styled.button
              isOperator={true}
            >
              <button
                className="sc-AxjAm ighVHx"
              >
                -
              </button>
            </styled.button>
            <styled.button>
              <button
                className="sc-AxjAm kVIsjG"
              >
                3
              </button>
            </styled.button>
            <styled.button>
              <button
                className="sc-AxjAm kVIsjG"
              >
                2
              </button>
            </styled.button>
            <styled.button>
              <button
                className="sc-AxjAm kVIsjG"
              >
                1
              </button>
            </styled.button>
            <styled.button
              isOperator={true}
            >
              <button
                className="sc-AxjAm ighVHx"
              >
                +
              </button>
            </styled.button>
            <Styled(styled.button)>
              <button
                className="sc-AxjAm sc-AxirZ ePprjk"
              >
                0
              </button>
            </Styled(styled.button)>
            <styled.button>
              <button
                className="sc-AxjAm kVIsjG"
              >
                .
              </button>
            </styled.button>
            <styled.button
              isOperator={true}
            >
              <button
                className="sc-AxjAm ighVHx"
              >
                =
              </button>
            </styled.button>
          </div>
        </styled.div>
      </ButtonGrid>
    </div>
  </styled.div>
</App>
`;

exports[`App App should render 1`] = `
<styled.div>
  <styled.div>
    0
  </styled.div>
  <ButtonGrid
    onChange={[Function]}
  />
</styled.div>
`;
